docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -996,48 +1068,219 @@ The UI design consists of three parts:
 - A place for users to manage their owned objects to be ready for escrow trading called `Manage Objects`.
 - A place for users to discover, create, and execute trades called `Escrows`.
 
-:::warning
-
-The following code snippets are not the full source code. The snippets are meant to focus on relevant logic important to the functionality of the example and features of Sui.
-
-:::
-
-### Set up providers {#set-up-providers}
-
-Set up and configure several providers at the root of your React.js tree to ensure different libraries including `dApp Kit`, `@tanstack/react-query`, `react-router-dom` work as expected.
-
-```ts title='src/main.tsx'
-import { createNetworkConfig, SuiClientProvider, WalletProvider } from '@mysten/dapp-kit';
-import { getFullnodeUrl } from '@mysten/sui/client';
-import { Theme } from '@radix-ui/themes';
-import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
-import { RouterProvider } from 'react-router-dom';
-
-import { router } from '@/routes/index.tsx';
-
-const queryClient = new QueryClient();
-
-const { networkConfig } = createNetworkConfig({
-	localnet: { url: getFullnodeUrl('localnet') },
-	devnet: { url: getFullnodeUrl('devnet') },
-	testnet: { url: getFullnodeUrl('testnet') },
-	mainnet: { url: getFullnodeUrl('mainnet') },
-});
+### Scaffold a new app
+
+The first step is to set up the client app. Run the following command to scaffold a new app.
+
+<Tabs groupId="packagemanager">
+  <TabItem label="PNPM" value="pnpm">
+    ```bash
+    pnpm create @mysten/dapp --template react-client-dapp
+    ```
+  </TabItem>
+  <TabItem label="Yarn" value="yarn">
+    ```bash
+    yarn create @mysten/dapp --template react-client-dapp
+    ```
+  </TabItem>
+</Tabs>
+
+### Setting up import aliases
+
+First, set up import aliases to make the code more readable and maintainable. This allows you to import files using `@/` instead of relative paths.
+
+<details>
+  <summary>
+    Replace the content of `tsconfig.json` with the following:
+  </summary>
+  {@inject: examples/trading/frontend/tsconfig.json}
```

Comment:
I'm seeing some strange behaviour w.r.t. indentation in some of the snippets, e.g. in this one we are getting rid of too much indentation.